src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -150,6 +150,19 @@ def validate_nsg_name_or_id(namespace):
                 type='networkSecurityGroups',
                 name=namespace.network_security_group)
 
+def validate_peering_type(namespace):
+    if namespace.peering_type and namespace.peering_type.lower() == 'microsoftpeering':
```

Comment:
why we need case insensitive checking? I think our `enum_choice_list` should have converted to the correct case so you can just compare with the regular case; otherwise, you should do the same for `create_express_route_peering` below